src/app/profile/edit-profile/edit-profile.component.ts

```diff
@@ -22,53 +25,116 @@ export class EditProfileComponent implements OnInit {
     firstName: " ",
     lastName: " ",
     team: " ",
-    skills: []
+    genericSkills: [],
+    domainSkills: [],
+    kbcSkills: []
   }
 
   profileForm: FormGroup;
   availableTeams: teams[];
-  userSkills: Skill[];
-  availableSkills: AvailableSkill[];
+  userGenericSkills: Skill[];
+  userDomainSkills: Skill[];
+  userKbcSkills: Skill[];
+  inputSkillArray: Skill[];
+
+  inputId: number;
+  maxSkillValue: number = 8;
+  typeToEdit: string;
+  availableGenericSkills: AvailableSkill[];
+  availableDomainSkills: AvailableSkill[];
+  availableKbcSkills: AvailableSkill[];
+  domLoaded: boolean = false;
 
   constructor(private dataStorageService: DataStorage,
               private fb: FormBuilder,
-              public dialog: MatDialog,) { }
+              private router: Router,
+              public dialog: MatDialog,
+              private activatedRoute: ActivatedRoute,
+              private bottomSheet: MatBottomSheet) { }
 
   ngOnInit() {
     this.initForm();
+    this.activatedRoute.params.subscribe(
+      (params : Params)=>{
+        this.inputId = +params['id'];
+      } 
+    );
+    this.activatedRoute.queryParams.subscribe(
+      (params : Params)=>{
+        this.typeToEdit = params['skill'];
+      } 
+    );
+    this.dataStorageService.getUser(this.inputId);
     this.dataStorageService.loggedInUser.subscribe(
       users=>{
           this.userData = users;
-          this.userSkills = this.userData.skills;
-          this.setForm(); 
-      }
+          this.userGenericSkills = this.userData.genericSkills;
+          this.userDomainSkills = this.userData.domainSkills;
+          this.userKbcSkills = this.userData.kbcSkills;
+          this.setForm();
+          this.domLoaded = true;
+            }
     );
     this.dataStorageService.getTeams().subscribe(teams=>{
       this.availableTeams = teams
     });  
-    this.dataStorageService.getSkills().subscribe(skills=>{
-      this.availableSkills = skills
+    this.dataStorageService.getGenericSkills().subscribe(skills=>{
+      this.availableGenericSkills = skills;
+    });
+    this.dataStorageService.getDomainSkills().subscribe(skills=>{
+      this.availableDomainSkills = skills;
+    });
+    this.dataStorageService.getKbcSkills().subscribe(skills=>{
+      this.availableKbcSkills = skills;
     });
   }  
 
-  onAddSkill(){
```

Comment:
All methods have same functionality.. just pass in the skill as a parameter alone in  a common method

Reply:
done